### PR TITLE
Bug 1945308: Backport lr nat del add fixes

### PR DESCRIPTION
**Pull request by trozet**
Separates lr-nat-del and lr-nat-add into two operations. This is required so we can safely bump OVN in 4.6z later.

### CODE REVIEW

**Comment by trozet** [comment]
@aojea @dcbw can you PTAL at this? Had a bunch of merge conflicts so just want to make sure it is reviewed carefully.

File: go-controller/pkg/ovn/egressgw.go [hunk 1]
```diff
@@ -307,22 +307,15 @@ func (oc *Controller) addPerPodGRSNAT(pod *kapi.Pod, podIfAddrs []*net.IPNet) er
 	}
 	gr := "GR_" + nodeName
 	for _, gwIPNet := range l3GWConfig.IPAddresses {
-		gwIP := gwIPNet.IP.String()
 		for _, podIPNet := range podIfAddrs {
```
Comment (dcbw) [inline go-controller/pkg/ovn/egressgw.go:310]: @trozet upstream has this bit too:

```
	for _, gwIPNet := range l3GWConfig.IPAddresses {
		gwIP := gwIPNet.IP.String()
		for _, podIPNet := range podIfAddrs {
			podIP := podIPNet.IP.String()
			if utilnet.IsIPv6String(gwIP) != utilnet.IsIPv6String(podIP) {
				continue
			}
			mask := GetIPFullMask(podIP)
```

Comment (dcbw) [reply go-controller/pkg/ovn/egressgw.go:310]: It's from https://github.com/ovn-org/ovn-kubernetes/pull/1944

Comment (dcbw) [reply go-controller/pkg/ovn/egressgw.go:310]: Tim reminds me that this is for 4.6, and we don't support v6 in 4.6, so this is not necessary.

**Comment by russellb** [comment]
last run failed to acquire a lease for infrastructure to run on

/test e2e-vsphere-ovn

**Comment by trozet** [comment]
Looks like there are docker pull limit problems...
```
[36mINFO[0m[2021-04-09T23:06:07Z] [It] should perform lookup when the Deployment gets the resolve-names annotation later [Suite:openshift/conformance/parallel] 
[36mINFO[0m[2021-04-09T23:06:07Z]   github.com/openshift/origin/test/extended/images/resolve.go:417 
[36mINFO[0m[2021-04-09T23:06:07Z] Apr  9 22:36:30.773: INFO: Running 'oc --namespace=e2e-test-resolve-local-names-4tv5r --kubeconfig=/tmp/configfile654160012 import-image busybox:latest --confirm' 
[36mINFO[0m[2021-04-09T23:06:07Z] error: tag latest failed: Internal error occurred: docker.io/library/busybox:latest: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```
Didn't we move to quay.io? Maybe that was in 4.7 and later? @dcbw any ideas?

**Pull request merged**